From bfcaffc73c01bcd53cdbf2bf7570b52deb4eef16 Mon Sep 17 00:00:00 2001
From: Zygmunt Krynicki <me@zygoon.pl>
Date: Mon, 21 Jan 2019 16:54:44 +0100
Subject: [PATCH 8/9] snap/squashsh: skip TestBuildDate on Debian

On Debian the reported build date of a squashfs file is about two days
back from what the test expected.

Signed-off-by: Zygmunt Krynicki <me@zygoon.pl>
---
 snap/squashfs/squashfs_test.go | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/snap/squashfs/squashfs_test.go b/snap/squashfs/squashfs_test.go
index 61613e7a0..70f52c234 100644
--- a/snap/squashfs/squashfs_test.go
+++ b/snap/squashfs/squashfs_test.go
@@ -35,6 +35,7 @@ import (
 
 	"github.com/snapcore/snapd/dirs"
 	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/release"
 	"github.com/snapcore/snapd/snap/snapdir"
 	"github.com/snapcore/snapd/snap/squashfs"
 	"github.com/snapcore/snapd/testutil"
@@ -590,6 +591,10 @@ func (s *SquashfsTestSuite) TestUnsquashfsStderrWriter(c *C) {
 }
 
 func (s *SquashfsTestSuite) TestBuildDate(c *C) {
+	if release.ReleaseInfo.ID == "debian" {
+		c.Skip("on Debian the reported build time was (presumably) that of source files")
+	}
+
 	// make a directory
 	d := c.MkDir()
 	// set its time waaay back
-- 
2.17.1

